src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -493,6 +493,9 @@ def _forward(self, model_inputs, return_timestamps=False, **generate_kwargs):
             else:
                 generate_kwargs["encoder_outputs"] = encoder(inputs, attention_mask=attention_mask)
 
+            if "assistant_model" in generate_kwargs:
+                generate_kwargs["input_features"] = inputs
```

Comment:
This logic should be under the `if` statement for encoder-decoder models only? i.e.
```python
if self.type in {"seq2seq", "seq2seq_whisper"}:
    ...
    if "input_features" in model_inputs:
        ...
    elif "input_values" in model_inputs:
        ...
    else:
        ...
    if "assistant_model" in generate_kwargs:
        generate_kwargs["input_features"] = inputs
    ...

Reply:
It would be a good exercise to figure out _why_ making this change works as well - if you could then explain on the PR description that would be grand